backend/src/controllers/squad.js

```diff
@@ -0,0 +1,101 @@
+exports.getSquad = async (req, res, next) => {  
+  try {
+    // Fetch squad from database
+    const squad = await req.context.models.Squad.findByPk(req.params.squadId);
+    const squadData = squad.dataValues;
+  
+    // Return squad
+    res.status(201);
+    res.send(squadData);
+  } catch (err) {
+    console.log(err)
+    res.status(404);
+    res.send({ error: "Could not find squad." });
+  }
+}
+
+exports.getSquads = async (req, res, next) => {
+  try {
+    // Fetch squads from database
+    const squads = await req.context.models.Squad.findAll({ where: { userId: req.user.id }});
+    
+    // Return No Content if the user has no squads
+    if (squads.length < 1) {
+      res.status(204);
+      return res.send({ message: "This user has no squads."})
+    }
+
+    // Return list of squads
+    res.status(200);
+    res.send(squads);
+  } catch (err) {
+    console.log(err);
+    res.status(404);
+    res.send({ error: "Could not find squads."});
+  }
+}
+
+exports.postSquad = async (req, res, next) => {
+  // Return Bad Request if data is missing
+  if (!req.body.name) {
+    return res.sendStatus(400);
+  }
+
+  const name = req.body.name;
+  const userId = req.user.id;
+
+  try {
+    const currentUser = await req.context.models.User.findByPk(userId);
+    
+    // Create Squad
+    const squad = await currentUser.createSquad({
+      name: name,
+      memberCount: 1
+    });
```

Comment:
What's the idea behind creating the squad on the user? Membership isn't implemented, and sequelize isn't generating any new tables.

Reply:
This isn't using sequelize to it's full potential though... https://sequelize.readthedocs.io/en/v3/api/associations/has-many/

The whole point of these association methods is that it auto builds them for you. In our example, a user HAS MANY squads. So a user can createSquad() and sequelize automatically connects the table for you.

When we start working with many : many relationships I would much rather have sequelize handle all the relation mapping then trying to hardcode it ourselves. 

Can you read the documentation and let me know if you're still not sold? Maybe try it with user and auth: https://sequelize.readthedocs.io/en/v3/api/associations/has-one/

Reply:
All good points! This works for me! I'll create a ticket to match auth/user to how you have it, for the 1:1. 

> Users and Squads are 1 : m not 1:1 unless I'm misunderstanding.

To clarify, I was referring to what's currently implemented in this pr. When you look in the database, all it's adding `userId` to the squads table. Which would represent a 1:1 relation (representing the creator of the squad). The way it's implemented currently there's no 1:m relation as far as I can see and test. 

As a sidenote, squads/users is m:m. One user has many squads. One squad has many users. No use or squad is uniquely tied to the other.

Reply:
hmmmmm, maybe I'm not understanding but what I was reading says otherwise: https://code.tutsplus.com/articles/sql-for-beginners-part-3-database-relationships--net-8561

There's an example of 1 to many where 1 customer has many orders. This is like one user being able to create many squads. no?

Reply:
Correct, my bad!